test/rekt/channel_test.go

```diff
@@ -301,3 +301,21 @@ func TestChannelPreferHeaderCheck(t *testing.T) {
 
 	env.Test(ctx, t, channel.ChannelPreferHeaderCheck(createSubscriberFn))
 }
+
+func TestChannelSubscriptionReturnedErrorData(t *testing.T) {
+	t.Parallel()
+
+	ctx, env := global.Environment(
+		knative.WithKnativeNamespace(system.Namespace()),
+		knative.WithLoggingConfig,
+		knative.WithTracingConfig,
+		k8s.WithEventListener,
+		environment.Managed(t),
+	)
+
+	createSubscriberFn := func(ref *duckv1.KReference, uri string) manifest.CfgFn {
+		return subscription.WithSubscriber(ref, uri)
+	}
+
+	env.Test(ctx, t, channel.ChannelSubscriptionReturnedErrorData(createSubscriberFn))
+}
```

Comment:
Should we additionally add a similar test that used the Broker/Trigger API?

Reply:
The extensions is not added in Broker/Trigger,  I am working on this PR https://github.com/knative/eventing/issues/6541

Reply:
Oh, I missed your comment https://github.com/knative/eventing/issues/6541#issuecomment-1278454349, thanks!